controller/Controller.class.php

```diff
@@ -198,4 +198,29 @@ public static function shutdown()
             call_user_func($fn);
         }
     }
+
+    protected static function doLowerUriRedirectOr404($uri)
+    {
+        $router = static::getRouterWithRoutes();
+        // $lowerUri = strtolower($uri);
+        $lowerUri = '/news/lbry-evolves';
+        if ($uri !== $lowerUri) {
+            if (View::exists('page/' . $lowerUri) || static::checkForRoute($lowerUri, $router)) {
+                return static::redirect($lowerUri, 308);
+            }
+        }
+        return NavActions::execute404();
+    }
+
+    protected static function checkForRoute($uri, $router)
+    {
+        $routerData = $router->getData();
+        if (in_array($uri, $routerData->getStaticRoutes())) {
```

Comment:
1) If this code was valid, it could simply be: `return in_array(xxx) || in_array(zzz)`
2) This code is not valid. You have to perform the actual routing to see if the URI matches, you can't simply do a key check.